test/metabase/cmd/dump_to_h2_test.clj

```diff
@@ -35,3 +45,38 @@
   (with-redefs [dump-to-h2/dump-to-h2! (constantly 1)
                 cmd/system-exit! identity]
     (is (= 1 (cmd/dump-to-h2 "file1")))))
+
+(defn- persistent-jdbcspec
+  "Return a jdbc spec for the specified `db-type` on the db `db-name`. In case of H2, makes the connection persistent
+  10secs to give us time to fetch the results later."
+  [db-type db-name app-db]
+  (case db-type
+    :h2 {:subprotocol "h2"
+         :subname     (format "mem:%s;DB_CLOSE_DELAY=10" app-db)
+         :classname   "org.h2.Driver"}
+    :postgres (db.spec/postgres (tx/dbdef->connection-details :postgres :db {:database-name db-name}))
+    :mysql (db.spec/mysql (tx/dbdef->connection-details :mysql :db {:database-name db-name}))))
+
+(defn- abs-path
+  [path]
+  (.getAbsolutePath (io/file path)))
+
+(deftest dump-to-h2-dump-plaintext-test
+  (let [h2-fixture-db-file (abs-path "frontend/test/__runner__/test_db_fixture.db")
+        app-db (mt/random-name)
+        h2-file (format "/tmp/out-%s.db" (mt/random-name))
+        db-name "test"]
+    (mt/test-drivers #{:h2 :postgres :mysql}
+     (binding [mdb.connection/*db-type*   driver/*driver*
+               mdb.connection/*jdbc-spec* (persistent-jdbcspec driver/*driver* db-name app-db)
+               db/*db-connection* (persistent-jdbcspec driver/*driver* db-name app-db)
+               db/*quoting-style* driver/*driver*]
+       (when-not (= driver/*driver* :h2)
+         (tx/create-db! driver/*driver* {:database-name db-name }))
+       (load-from-h2/load-from-h2! h2-fixture-db-file)
+       (eu/with-secret-key "89ulvIGoiYw6mNELuOoEZphQafnF/zYe+3vT+v70D1A="
+         (db/insert! Setting {:key "my-site-admin", :value "baz"})
+         (dump-to-h2/dump-to-h2! h2-file {:dump-plaintext? true}))
```

Comment:
might be worth adding a test with `:dump-plaintext? false` here so you can see if the Setting is still encrypted in that version

Reply:
you should probably also test Database `details` here as well

Reply:
added tests for details and set flag to false as well as test that default==false